### ISSUE TITLE
Docs: [ban-ts-comment] ts-expect-error option description is inconsistent

### ISSUE THREAD

**Issue by zaewc** [typescript-eslint/typescript-eslint#12464]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [x] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The `ts-expect-error` option's schema `description` in `ban-ts-comment` is broken and inconsistent with its sibling directive options

All four directive options should share the same sentence pattern, but `ts-expect-error` doesn't:

| Option | `description` |
| --- | --- |
| `ts-check` | `Whether allow ts-check directives, and with which restrictions.` |
| `ts-ignore` | `Whether allow ts-ignore directives, and with which restrictions.` |
| `ts-nocheck` | `Whether allow ts-nocheck directives, and with which restrictions.` |
| **`ts-expect-error`** | **`Whether and when expect-error directives, and with which restrictions.`** |

There are two problems with the `ts-expect-error` description (`packages/eslint-plugin/src/rules/ban-ts-comment.ts`):

1. **The verb is missing.** "Whether and when expect-error directives, …" isn't a grammatical sentence — the `allow` verb that the other three have is gone
2. **The `ts-` prefix is missing.** It says `expect-error` while the siblings say `ts-check` / `ts-ignore` / `ts-nocheck`. The directive is `@ts-expect-error`, so it should read `ts-expect-error`

These descriptions are surfaced to users both in the auto-generated Options section of the rule's docs page and as tooltips in editor autocomplete for the ESLint config, so the broken wording is visible

### Affected URL(s)

https://typescript-eslint.io/rules/ban-ts-comment/#options

**Comment by zaewc** [comment]
Suggested fix, to keep it parallel with the other three options:

```ts
'Whether allow ts-expect-error directives, and with which restrictions.'
```

**Comment by kirkwaiblinger** [comment]
should probably fix the grammar in "whether allow" too 👍
